plugins/rebalance/process.go

```diff
@@ -0,0 +1,110 @@
+)
+
+const (
+	glusterfsBin = "glusterfs"
+)
+
+// Process type represents information about rebalance process
+type Process struct {
+	binarypath     string
+	args           string
+	socketfilepath string
+	pidfilepath    string
+	rInfo          rebalanceapi.RebalInfo
+}
+
+// Name returns the process name
+func (r *Process) Name() string {
+	return "rebalance"
+}
+
+// Path returns absolute path to the binary of rebalance process
+func (r *Process) Path() string {
+	return r.binarypath
+}
+
+// Args returns arguments to be passed to rebalance process
+func (r *Process) Args() string {
+
+	volfileserver, port, _ := net.SplitHostPort(config.GetString("clientaddress"))
+	if volfileserver == "" {
+		volfileserver = "localhost"
+	}
+
+	volFileID := path.Join("rebalance/", r.rInfo.Volname)
+	logDir := path.Join(config.GetString("logdir"), "glusterfs")
+	logFile := fmt.Sprintf("%s/%s-rebalance.log", logDir, r.rInfo.Volname)
+	cmd := r.rInfo.Cmd
+	commithash := r.rInfo.CommitHash
+
+	var buffer bytes.Buffer
+	buffer.WriteString(fmt.Sprintf(" -s %s --volfile-server-port %s", volfileserver, port))
+	buffer.WriteString(fmt.Sprintf(" --volfile-id %s", volFileID))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.use-readdirp=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.lookup-unhashed=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.assert-no-child-down=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.data-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.metadata-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.entry-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.readdir-optimize=on"))
+	buffer.WriteString(fmt.Sprintf(" --process-name rebalance"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.rebalance-cmd=%d", cmd))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.node-uuid=%s", gdctx.MyUUID))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *distribute.commit-hash=%d", commithash))
+	buffer.WriteString(fmt.Sprintf(" -p %s", r.PidFile()))
+	buffer.WriteString(fmt.Sprintf(" --socket-file %s", r.SocketFile()))
+	buffer.WriteString(fmt.Sprintf(" -l %s", logFile))
+	r.args = buffer.String()
+	return r.args
+}
+
+// SocketFile returns path to the socket file used for IPC
+func (r *Process) SocketFile() string {
+
+	if r.socketfilepath != "" {
+		return r.socketfilepath
+	}
+
+	glusterdSockDir := path.Join(config.GetString("rundir"), "", r.rInfo.Volname)
+	r.socketfilepath = fmt.Sprintf("%s-rebalance-%x.socket", glusterdSockDir, xxhash.Sum64String(gdctx.MyUUID.String()))
+	return r.socketfilepath
+}
+
+// PidFile returns path to the pid file of rebalance process
+func (r *Process) PidFile() string {
+	if r.pidfilepath != "" {
+		return r.pidfilepath
+	}
+
+	r.pidfilepath = fmt.Sprintf("%s/%s-rebalance.pid", config.GetString("rundir"), r.rInfo.Volname)
```

Comment:
pid file is used as `<volname>-rebalance`, ID below is used as `rebalance-<volname>`

Reply:
Doesn't actually matter much as long as each is unique. But it is nice to have them similar.